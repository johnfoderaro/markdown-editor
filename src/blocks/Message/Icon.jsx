import React from 'react';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Loader = styled.svg`
  margin-left: auto;
  margin-right: auto;
  transform: rotate(-90deg);  
  stroke: #fff;
  circle {
    stroke-dasharray: 260;
    stroke-dashoffset: 260;
    animation: dash 1.5s infinite;  
  }
  @keyframes dash {
    50% {
      stroke-dashoffset: 0;
    }
    100% {
      stroke-dashoffset: -260;
    }  
  }
`;

const Svg = styled.svg`
  margin-left: auto;
  margin-right: auto;
`;

const Icon = ({ type }) => {
  if (type === 'error') {
    return (
      <Svg width="120" height="120" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
        <path d="M1440 893q0-161-87-295l-754 753q137 89 297 89 111 0 211.5-43.5t173.5-116.5 116-174.5 43-212.5zm-999 299l755-754q-135-91-300-91-148 0-273 73t-198 199-73 274q0 162 89 299zm1223-299q0 157-61 300t-163.5 246-245 164-298.5 61-298.5-61-245-164-163.5-246-61-300 61-299.5 163.5-245.5 245-164 298.5-61 298.5 61 245 164 163.5 245.5 61 299.5z" fill="#fff" />
      </Svg>
    );
  }
  if (type === 'warning') {
    return (
      <Svg width="120" height="120" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
        <path d="M1008 1200v160q0 14-9 23t-23 9h-160q-14 0-23-9t-9-23v-160q0-14 9-23t23-9h160q14 0 23 9t9 23zm256-496q0 50-15 90t-45.5 69-52 44-59.5 36q-32 18-46.5 28t-26 24-11.5 29v32q0 14-9 23t-23 9h-160q-14 0-23-9t-9-23v-68q0-35 10.5-64.5t24-47.5 39-35.5 41-25.5 44.5-21q53-25 75-43t22-49q0-42-43.5-71.5t-95.5-29.5q-56 0-95 27-29 20-80 83-9 12-25 12-11 0-19-6l-108-82q-10-7-12-20t5-23q122-192 349-192 129 0 238.5 89.5t109.5 214.5zm-368-448q-130 0-248.5 51t-204 136.5-136.5 204-51 248.5 51 248.5 136.5 204 204 136.5 248.5 51 248.5-51 204-136.5 136.5-204 51-248.5-51-248.5-136.5-204-204-136.5-248.5-51zm768 640q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z" fill="#fff" />
      </Svg>
    );
  }
  if (type === 'success') {
    return (
      <Svg width="120" height="120" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
        <path d="M1299 813l-422 422q-19 19-45 19t-45-19l-294-294q-19-19-19-45t19-45l102-102q19-19 45-19t45 19l147 147 275-275q19-19 45-19t45 19l102 102q19 19 19 45t-19 45zm141 83q0-148-73-273t-198-198-273-73-273 73-198 198-73 273 73 273 198 198 273 73 273-73 198-198 73-273zm224 0q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z" fill="#fff" />
      </Svg>
    );
  }
  if (type === 'loader') {
    return (
      <Loader height="120" width="120">
        <circle cx="60" cy="60" r="40" strokeWidth="10" fill="none" />
      </Loader>
    );
  }
  return false;
};

Icon.propTypes = {
  type: PropTypes.string,
};


export default hot(module)(Icon);
