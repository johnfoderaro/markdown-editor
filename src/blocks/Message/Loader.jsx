import React from 'react';
import { hot } from 'react-hot-loader';
import styled from 'styled-components';

const Svg = styled.svg`
  margin-left: auto;
  margin-right: auto;
  transform: rotate(-90deg);
  stroke: #fff;
  circle {
    stroke-dasharray: 260;
    stroke-dashoffset: 260;
    animation: dash 1.5s infinite;  
  }
  @keyframes dash {
    50% {
      stroke-dashoffset: 0;
    }
    100% {
      stroke-dashoffset: -260;
    }  
  }
`;

const Loader = () => (
  <Svg height="120" width="120">
    <circle cx="60" cy="60" r="40" strokeWidth="10" fill="none" />
  </Svg>
);

export default hot(module)(Loader);
