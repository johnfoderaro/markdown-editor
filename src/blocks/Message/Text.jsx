import styled from 'styled-components';

const Text = styled.span`
  padding: 1rem;
  color: #fff;
  font-size: 1.5em;
`;

export default Text;
