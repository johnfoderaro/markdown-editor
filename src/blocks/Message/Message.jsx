import styled from 'styled-components';

import Text from './Text';
import Error from './Error';
import Warning from './Warning';
import Success from './Success';
import Loader from './Loader';

const Message = styled.div`
  display: flex;
  flex-direction: column;
`;

Message.Text = Text;
Message.Error = Error;
Message.Warning = Warning;
Message.Success = Success;
Message.Loader = Loader;

export default Message;
