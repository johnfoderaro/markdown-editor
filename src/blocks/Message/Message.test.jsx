import React from 'react';
import { shallow } from 'enzyme';

import Message from './Message';

let tree;

beforeAll(() => {
  tree = shallow(
    <Message>
      <h1>one</h1>
      <p>two</p>
    </Message>,
  );
});

describe('<Message />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should render children correctly', () => {
    expect(tree.find('h1').text()).toEqual('one');
    expect(tree.find('p').text()).toEqual('two');
  });
});
