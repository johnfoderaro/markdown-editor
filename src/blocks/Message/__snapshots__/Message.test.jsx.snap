// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render correctly 1`] = `
.c1 {
  margin-left: auto;
  margin-right: auto;
}

.c2 {
  margin-left: auto;
  margin-right: auto;
}

.c5 {
  margin-left: auto;
  margin-right: auto;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  stroke: #fff;
}

.c5 circle {
  stroke-dasharray: 260;
  stroke-dashoffset: 260;
  -webkit-animation: dash 1.5s infinite;
  animation: dash 1.5s infinite;
}

.c4 {
  margin-left: auto;
  margin-right: auto;
}

.c3 {
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "Bomb": [Function],
        "Error": [Function],
        "Loader": [Function],
        "Success": [Function],
        "Text": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  padding: 1rem;
  color: #fff;
  font-size: 1.5em;
",
            ],
          },
          "displayName": "styled.span",
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "span",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Warning": [Function],
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      <Bomb>
        <styled.svg
          height="120"
          viewBox="0 0 1792 1792"
          width="120"
          xmlns="http://www.w3.org/2000/svg"
        >
          <StyledComponent
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  margin-left: auto;
  margin-right: auto;
",
                  ],
                },
                "displayName": "styled.svg",
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "svg",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height="120"
            viewBox="0 0 1792 1792"
            width="120"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              className="c1"
              height="120"
              viewBox="0 0 1792 1792"
              width="120"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M571 589q-10-25-34-35t-49 0q-108 44-191 127t-127 191q-10 25 0 49t35 34q13 5 24 5 42 0 60-40 34-84 98.5-148.5t148.5-98.5q25-11 35-35t0-49zm942-356l46 46-244 243 68 68q19 19 19 45.5t-19 45.5l-64 64q89 161 89 343 0 143-55.5 273.5t-150 225-225 150-273.5 55.5-273.5-55.5-225-150-150-225-55.5-273.5 55.5-273.5 150-225 225-150 273.5-55.5q182 0 343 89l64-64q19-19 45.5-19t45.5 19l68 68zm8-56q-10 10-22 10-13 0-23-10l-91-90q-9-10-9-23t9-23q10-9 23-9t23 9l90 91q10 9 10 22.5t-10 22.5zm230 230q-11 9-23 9t-23-9l-90-91q-10-9-10-22.5t10-22.5q9-10 22.5-10t22.5 10l91 90q9 10 9 23t-9 23zm41-183q0 14-9 23t-23 9h-96q-14 0-23-9t-9-23 9-23 23-9h96q14 0 23 9t9 23zm-192-192v96q0 14-9 23t-23 9-23-9-9-23v-96q0-14 9-23t23-9 23 9 9 23zm151 55l-91 90q-10 10-22 10-13 0-23-10-10-9-10-22.5t10-22.5l90-91q10-9 23-9t23 9q9 10 9 23t-9 23z"
                fill="#fff"
              />
            </svg>
          </StyledComponent>
        </styled.svg>
      </Bomb>
      <Error>
        <styled.svg
          height="120"
          viewBox="0 0 1792 1792"
          width="120"
          xmlns="http://www.w3.org/2000/svg"
        >
          <StyledComponent
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  margin-left: auto;
  margin-right: auto;
",
                  ],
                },
                "displayName": "styled.svg",
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "svg",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height="120"
            viewBox="0 0 1792 1792"
            width="120"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              className="c2"
              height="120"
              viewBox="0 0 1792 1792"
              width="120"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1440 893q0-161-87-295l-754 753q137 89 297 89 111 0 211.5-43.5t173.5-116.5 116-174.5 43-212.5zm-999 299l755-754q-135-91-300-91-148 0-273 73t-198 199-73 274q0 162 89 299zm1223-299q0 157-61 300t-163.5 246-245 164-298.5 61-298.5-61-245-164-163.5-246-61-300 61-299.5 163.5-245.5 245-164 298.5-61 298.5 61 245 164 163.5 245.5 61 299.5z"
                fill="#fff"
              />
            </svg>
          </StyledComponent>
        </styled.svg>
      </Error>
      <Warning>
        <styled.svg
          height="120"
          viewBox="0 0 1792 1792"
          width="120"
          xmlns="http://www.w3.org/2000/svg"
        >
          <StyledComponent
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-EHOje",
                  "isStatic": true,
                  "lastClassName": "c3",
                  "rules": Array [
                    "
  margin-left: auto;
  margin-right: auto;
",
                  ],
                },
                "displayName": "styled.svg",
                "render": [Function],
                "styledComponentId": "sc-EHOje",
                "target": "svg",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height="120"
            viewBox="0 0 1792 1792"
            width="120"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              className="c3"
              height="120"
              viewBox="0 0 1792 1792"
              width="120"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1008 1200v160q0 14-9 23t-23 9h-160q-14 0-23-9t-9-23v-160q0-14 9-23t23-9h160q14 0 23 9t9 23zm256-496q0 50-15 90t-45.5 69-52 44-59.5 36q-32 18-46.5 28t-26 24-11.5 29v32q0 14-9 23t-23 9h-160q-14 0-23-9t-9-23v-68q0-35 10.5-64.5t24-47.5 39-35.5 41-25.5 44.5-21q53-25 75-43t22-49q0-42-43.5-71.5t-95.5-29.5q-56 0-95 27-29 20-80 83-9 12-25 12-11 0-19-6l-108-82q-10-7-12-20t5-23q122-192 349-192 129 0 238.5 89.5t109.5 214.5zm-368-448q-130 0-248.5 51t-204 136.5-136.5 204-51 248.5 51 248.5 136.5 204 204 136.5 248.5 51 248.5-51 204-136.5 136.5-204 51-248.5-51-248.5-136.5-204-204-136.5-248.5-51zm768 640q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
                fill="#fff"
              />
            </svg>
          </StyledComponent>
        </styled.svg>
      </Warning>
      <Success>
        <styled.svg
          height="120"
          viewBox="0 0 1792 1792"
          width="120"
          xmlns="http://www.w3.org/2000/svg"
        >
          <StyledComponent
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "lastClassName": "c4",
                  "rules": Array [
                    "
  margin-left: auto;
  margin-right: auto;
",
                  ],
                },
                "displayName": "styled.svg",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "svg",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height="120"
            viewBox="0 0 1792 1792"
            width="120"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              className="c4"
              height="120"
              viewBox="0 0 1792 1792"
              width="120"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1299 813l-422 422q-19 19-45 19t-45-19l-294-294q-19-19-19-45t19-45l102-102q19-19 45-19t45 19l147 147 275-275q19-19 45-19t45 19l102 102q19 19 19 45t-19 45zm141 83q0-148-73-273t-198-198-273-73-273 73-198 198-73 273 73 273 198 198 273 73 273-73 198-198 73-273zm224 0q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
                fill="#fff"
              />
            </svg>
          </StyledComponent>
        </styled.svg>
      </Success>
      <Loader>
        <styled.svg
          height="120"
          width="120"
        >
          <StyledComponent
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "c5",
                  "rules": Array [
                    "
  margin-left: auto;
  margin-right: auto;
  transform: rotate(-90deg);
  stroke: #fff;
  circle {
    stroke-dasharray: 260;
    stroke-dashoffset: 260;
    animation: dash 1.5s infinite;  
  }
  @keyframes dash {
    50% {
      stroke-dashoffset: 0;
    }
    100% {
      stroke-dashoffset: -260;
    }  
  }
",
                  ],
                },
                "displayName": "styled.svg",
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "svg",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            height="120"
            width="120"
          >
            <svg
              className="c5"
              height="120"
              width="120"
            >
              <circle
                cx="60"
                cy="60"
                fill="none"
                r="40"
                strokeWidth="10"
              />
            </svg>
          </StyledComponent>
        </styled.svg>
      </Loader>
    </div>
  </StyledComponent>
</styled.div>
`;
